packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1053,12 +1072,24 @@ abstract class ParentDataWidget<T extends RenderObjectWidget> extends ProxyWidge
 /// Inherited widgets, when referenced in this way, will cause the consumer to
 /// rebuild when the inherited widget itself changes state.
 abstract class InheritedWidget extends ProxyWidget {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
   const InheritedWidget({ Key key, Widget child })
     : super(key: key, child: child);
 
   @override
   InheritedElement createElement() => new InheritedElement(this);
 
+  /// Whether the framework should notify widgets that inherit from this widget.
+  ///
+  /// When this widget is rebuilt, some new instances will require rebuilding
+  /// the widgets that inherit from this widget but other new instances do not
+  /// (e.g., if the data held by this widget is the same as the data held by the
+  /// previous instance of this widget that occupied this location in the tree).
```

Comment:
well, i may have had the headache already. but i felt it more when reading this sentence?